app/javascript/components/pages/SearchNew.js

```diff
@@ -41,14 +44,61 @@ class SearchNew extends React.Component {
         }
     }
 
+    findCities = (cityData) => {
+      let {form} = this.state
+      // console.log(cityData[0]);
+      let topCity = ""
+      let secondCity = ""
+      let thirdCity = ""
+      let topValue = 0.0
+      let secondValue = 0.0
+      let thirdValue = 0.0
+      console.log(this.searchKey)
+      console.log(this.searchValue)
+      cityData.filter(value => {
+        for (var i in value) {
+          if (i == this.searchKey) {
+            if (value[i] > this.searchValue) {
```

Comment:
It's a little difficult to follow what's happening here. I think renaming some of these variables might be an easy way to improve readability. Maybe renaming `value` to `city`  and `i` to `key` or `field`, etc. 

I think also, pulling this filtering function out into a separate function that takes the `searchValue` and `searchKey` and returns `topCity`, `secondCity`, and `thirdCity`. 

Lastly, if you have time to write a test for that function, I think this is good place for testing. It's a little difficult to read and there's a lot of variable mutation going on, which can generally be a source of bugs. You could follow up later with this if you don't have time, but if you do think there will be changes made to this logic in the course of the development, having tests around it is going to make those changes easier and faster.